packages/tsx-editor/src/transpiler/exampleTransform.ts

```diff
@@ -0,0 +1,92 @@
+export interface ITransformedExample {
+  output?: string;
+  error?: string;
+}
+
+export function transformExample(example: string, id: string) {
+  /**
+   * classNamePattern - pattern to find the name of the class that should render
+   * constNamePattern - pattern to find the name of the const that should render
+   * identifierPattern - pattern to get all identifiers from the imports
+   * importPattern - pattern to get all imports even if they have multilines
+   */
+  const classNamePattern = new RegExp('(?<=var )(.*)(?= = /\\*\\* @class \\*/ \\(function \\(_super)', 'g');
+  const constNamePattern = new RegExp('(?<=var )(.*)(?= = function())', 'g');
+  const identifierPattern = new RegExp('(?<=import {)([\\s\\S]*)(?=})');
+  const importPattern = new RegExp("(?:import)([\\s\\S]*?)(?:';)", 'g');
+  const identifiers: string[] = [];
+  const imports: string[] = [];
+  let temp;
+  let className;
+  const output: ITransformedExample = { output: undefined, error: undefined };
+
+  example = example.replace("import * as React from 'react';", '');
+
+  /**
+   * Getting class name that should render, it will find all class names but will only
+   * use the last one since the examples have the class that should render at the end.
+   * If there is no such class then it will check for the last const which should be
+   * the const that should be the one to render.
+   */
+  while ((temp = classNamePattern.exec(example))) {
+    className = temp[0];
+  }
+  if (className === undefined) {
+    while ((temp = constNamePattern.exec(example))) {
+      className = temp[0];
+    }
+  }
+
+  /**
+   * Getting all the imports from fabric and pushing them into an array.
+   * If there are imports that do not come from OUFR then it will add an
+   * error since the import is not supported.
+   */
+  while ((temp = importPattern.exec(example))) {
+    if (!/office-ui-fabric-react/.test(temp[0])) {
```

Comment:
Should probably ensure it's at the beginning of the string:
```suggestion
    if (!/^office-ui-fabric-react/.test(temp[0])) {
```

Reply:
it shouldn't matter, it just checks if that portion exists in the import.